lib/src/klio/transforms/io.py

```diff
@@ -151,47 +149,52 @@ def __iter__(self):
         # NOTE: We need to have the row elements be bytes, so if it is
         # a dictionary, we json.dumps into a str to convert to bytes,
         # but that may need to change if we want to support other coders
-        for row in super(_KlioBigQueryReader, self).__iter__():
-            message = self.__generate_klio_message()
+        data = {}
+        if self.__klio_message_columns:
+            if len(self.__klio_message_columns) == 1:
+                data = row[self.__klio_message_columns[0]]
 
-            data = {}
-            if self.__klio_message_columns:
-                if len(self.__klio_message_columns) == 1:
-                    data = row[self.__klio_message_columns[0]]
+            else:
+                for key, value in row.items():
+                    if key in self.__klio_message_columns:
+                        data[key] = value
+                data = json.dumps(data)
 
-                else:
-                    for key, value in row.items():
-                        if key in self.__klio_message_columns:
-                            data[key] = value
-                    data = json.dumps(data)
+        else:
+            data = json.dumps(row)
+        return data
 
-            else:
-                data = json.dumps(row)
+    def _map_row(self, row):
+        message = self._generate_klio_message()
+        message.data.element = bytes(self._map_row_element(row), "utf-8")
+        return message.SerializeToString()
 
-            message.data.element = bytes(data, "utf-8")
-            yield message.SerializeToString()
+    def as_beam_map(self):
```

Comment:
If the class itself is meant to be public (re: comment above), can you add a docstring for this one since it's a public method?